### PR TITLE
Containerize development environment

### PR DESCRIPTION
**Pull request by tmc**
## Status

Work in progress

## Description of Changes
This supplies the ability to build development environments via docker. This simplifies getting started for new developers as well as allows development on Qubes. Based on #2511 

## Testing

make app-images app-dev

This brings up the source and journalist interfaces on :8080, :8081

## Deployment
N/A

## Checklist
N/A

### CODE REVIEW

File: securedrop/bin/test [hunk 1]
```diff
@@ -3,7 +3,7 @@ set -euo pipefail
 set -x
 export DISPLAY=:1
 Xvfb :1 -screen 0 1024x768x24 -ac +extension GLX +render -noreset &
-haveged &
+haveged & haveged & haveged & haveged &
```
Comment (redshiftzero) [inline securedrop/bin/test]: :joy:

**Comment by tmc** [comment]
@redshiftzero with a pkill of rqworker a `DISPLAY=:1 pytest -v -s -x` just ran successfully for me. The issue there is related to #2613 in that the conftest-started rqworker refuses to start and the test queue isn't consumed. I can write docs.

I consider testinfra against containerized dev env a separate problem to solve. Separate Issue/PR?

**Comment by redshiftzero** [comment]
As we discussed at the hackathon, I think @msheiny's suggestion is reasonable. Approving :)

**Comment by pierwill** [comment]
Hello all, just peeking in from over in #2711. @tmc:  Once this is merged, I'd love to help out testing and documenting a Qubes SD virtual environment. I'm sure if I can get it set up, anyone can.

**Comment by dachary** [comment]
@msheiny I believe your concerns have been addressed ? If not could you explain what's left to do ?

**Pull request merged**